src/main/java/co/cask/hydrator/plugin/batch/HTTPSink.java

```diff
@@ -0,0 +1,402 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Sink plugin to send the messages from the pipeline to an external http endpoint.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name("HTTP")
+@Description("Sink plugin to send the messages from the pipeline to an external http endpoint.")
+
+public class HTTPSink extends ReferenceBatchSink<StructuredRecord, Void, Void> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HTTPSink.class);
+  private static final String KV_DELIMITER = ":";
+  private static final String DELIMITER = "\n";
+  private static final Set<String> METHODS = ImmutableSet.of(HttpMethod.GET, HttpMethod.POST,
+                                                             HttpMethod.PUT, HttpMethod.DELETE);
+  private static StringBuilder messages = new StringBuilder("");
+
+  private HTTPSinkConfig config;
+
+  public HTTPSink(HTTPSinkConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    config.validate();
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    context.addOutput(Output.of(config.referenceName, new HTTPSink.HTTPOutputFormatProvider()));
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<Void, Void>> emitter) throws Exception {
+    config.validate();
+    String message = null;
+    String contentType = null;
+    int responseCode;
+    int retries = 0;
+    Exception exception = null;
+
+    if (config.method.equals("POST") || config.method.equals("PUT")) {
+      if (config.messageFormat.equals("JSON")) {
+        message = StructuredRecordStringConverter.toJsonString(input);
+        contentType = "application/json";
+      } else if (config.messageFormat.equals("Form")) {
+        message = createFormMessage(input);
+        contentType = " application/x-www-form-urlencoded";
+      } else if (config.messageFormat.equals("Custom")) {
+        message = createCustomMessage(config.body, input);
+        contentType = "application/plain";
```

Comment:
In case user doesnt add a header property for content type,we would need this as "text/plain" for custom.